Service/FederalizationiService.php

```diff
@@ -0,0 +1,283 @@
+                $source =  new Source();
+                $source->setName($baseSync['source']['name']);
+                $source->setDescription($baseSync['source']['description']);
+                $source->setLocation($baseSync['source']['location']);
+            }
+        }
+        else{
+            // This catalogi is teh source so lets roll
+            $externalId = $object['id'];
+        }
+
+        // Lets se if we already have an synchronisation
+        if(!$synchonization = $this->entityManager->getRepository('App:Synchronization')->findOneBy(['sourceId' =>$externalId])){
+            $synchonization = new Synchronization($source, $entity);
+            $synchonization->setSourceId($object['id']);
+        }
+
+        if(isset($object['_self'])){ unset($object['_self']);}
+
+        // Lets sync
+        return $this->synchronizationService->handleSync($synchonization, $object);
+    }
+
+    /**
+     * Makes sure that we have the object entities that we need
+     *
+     * @return void
+     */
+    public function prebObjectEntities():void{
+        if(!isset($this->catalogusEntity)){
+            $this->catalogusEntity = $this->entityManager->getRepository('App:Entity')->findOneBy(['reference' =>'https://opencatalogi.nl/catalogi.schema.json']);
+            (!$this->applicationEntity && isset($this->io)?$this->io->error('Could not find a entity for https://opencatalogi.nl/catalogi.schema.json'):'');
+        }
+        if(!isset($this->componentEntity)){
+            $this->componentEntity = $this->entityManager->getRepository('App:Entity')->findOneBy(['reference' =>'https://opencatalogi.nl/component.schema.json']);
+            (!$this->applicationEntity && isset($this->io)?$this->io->error('Could not find a entity for https://opencatalogi.nl/component.schema.json'):'');
+        }
+        if(!isset($this->organisationEntity)){
+            $this->organisationEntity = $this->entityManager->getRepository('App:Entity')->findOneBy(['reference' =>'https://opencatalogi.nl/organisation.schema.json']);
+            (!$this->applicationEntity && isset($this->io)?$this->io->error('Could not find a entity for https://opencatalogi.nl/organisation.schema.json'):'');
+        }
+        if(!isset($this->applicationEntity)){
+            $this->applicationEntity = $this->entityManager->getRepository('App:Entity')->findOneBy(['reference' =>'https://opencatalogi.nl/application.schema.json']);
+            (!$this->applicationEntity && isset($this->io)?$this->io->error('Could not find a entity for https://opencatalogi.nl/application.schema.json'):'');
+        }
+    }
+
+    /**
+     * Handles the sync all catalogi action from the catalogi handler
+     *
+     * @param array $data
+     * @param array $configuration
+     * @return array
+     */
+    public function catalogiHandler(array $data = [], array $configuration = []): array{
+
+        // Setup base data
+        $this->prebObjectEntities();
+
+        // Savety cheek
```

Comment:
Moeten alle 4 entities niet gecheckt worden en zou die check naar prebObjectEntities kunnen

Reply:
Check zit daar nu ook zowiezo, maar in dit geval wil je hier de functie beindigen zodat het proces verder kan. Dus wil je hem wel nogmaals checken